src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -184,61 +221,42 @@ private function adaptResourceLocators(ResourceSegmentBehavior $document, $userI
         $webspaceKey = $this->documentInspector->getWebspace($document);
         $languageCode = $this->documentInspector->getLocale($document);
 
-        foreach ($document->getChildren() as $childDocument) {
-            $childNode = $this->documentInspector->getNode($childDocument);
-
-            // determine structure
-            $templatePropertyName = $this->nodeHelper->getTranslatedPropertyName('template', $languageCode);
+        $node = $this->documentInspector->getNode($document);
+        $node->getSession()->save();
 
-            if (!$childNode->hasProperty($templatePropertyName)) {
+        foreach ($document->getChildren() as $childDocument) {
+            // skip documents without assigned resource segment
+            if (!$childDocument instanceof ResourceSegmentBehavior
+                || !$currentResourceLocator = $childDocument->getResourceSegment()) {
```

Comment:
I wouldn't do the assignment for `$currentResourceLocator` in the if, that's quite hard to read. I think I would prefer the `adaptResourceLocators` call twice with the continue.